radixdlt/babylon-wallet-android
# App compatibility with sargon os branch

## Description
This PR aims to achieve compatibility with sargon os branch. 
> [!Warning]
> It is not booting sargon os yet, but makes our app compatible with some structural changes on https://github.com/radixdlt/sargon/pull/176

### Changes
1. More factor sources are included in this sargon branch but not operational. Exhaustive `when`s are handled appropriately. Code that usually produces a side effect, does nothing for new factor sources, and code that needs to return a value, returns a Result.Failure with an appropriate exception. Keep in mind that such errors will not be thrown since these factor sources are not operational.
2. Our object `DeviceInfo` was replaced with `HostId` and `HostInfo`. For more information related to those 2 objects in sargon take a look at this pr https://github.com/radixdlt/sargon/pull/189. In general the id and timestamp it was generated are the only information needed to be preserved (used heavily in the backup feature). The rest can be retrieved on the fly using `context` or `Build` information. Check this test `profile/src/test/java/rdx/works/profile/data/HostInfoRepositoryTest.kt` that tests the compatibility between the previous and the new object that preserves the id of the device.
3. Some object renamings

## How to test
1. The update should not affect the app. Everything should be operational.
2. One good test on top of the rest is to have a version of the app with backups enabled. Then update to this version and check that the screen that informs the user that the device was claimed by another device **was not shown** and backups continue without any problem.

This branch will be merged when https://github.com/radixdlt/sargon/pull/176 is merged to main
